### PR TITLE
AfluentTesting target

### PR DESCRIPTION
**Pull request by aim2120**
## Description

Adds an AfluentTesting target that contains some of the test functions that we'd like to make available publicly.

### AfluentTesting

Adds the `wait(until:timeout)` and `Task.waitUntilScheduled` test utilities to a new AfluentTesting target.

For `waitUntilScheduled`, I changed `sending @isolated(any)` to `@Sendable`, which is supported by Swift < 6 and Swift 6 compilers. @Tyler-Keith-Thompson if there's a particular reason why you used the former, let me know, because we'd need to use a conditional compiler directive to support Swift < 6.

### DocC Publishing Changes

Adds DocC publishing for the AfluentTesting target, in addition to the Afluent target.

I did this by using a [reusable workflow](https://docs.github.com/en/actions/sharing-automations/reusing-workflows). Open to suggestions on if there's a more preferred way to achieve the same effect.

I've temporarily added PR releasing to ensure the publishing works as expected.
See the docs published here:
- [Afluent](https://tyler-keith-thompson.github.io/Afluent/documentation/afluent/)
- [AfluentTesting](https://tyler-keith-thompson.github.io/Afluent/documentation/afluenttesting/)

I also went ahead and updated the runner to macos-15, since it's available.

### README

Added a few changes to the README:
- Links to both Afluent and AfluentTesting docs
- New "Feature" bullet
- How to add the Afluent and AfluentTesting targets

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -3,27 +3,20 @@ name: Release
 on:
   release:
     types: [ created ]
+  pull_request:
```
Comment (aim2120) [inline .github/workflows/release.yml:6]: Revert this change before merging

File: Sources/AfluentTesting/WaitUntilScheduled.swift [hunk 2]
```diff
@@ -9,8 +9,8 @@ import Afluent
 
 extension Task where Failure == Error {
     /// Spawns a new Task to run some async operation and waits for that task to begin execution before proceeding.
-    static func waitUntilScheduled(
-        operation: sending @escaping @isolated(any) () async throws -> Success
+    public static func waitUntilScheduled(
```
Comment (Tyler-Keith-Thompson) [inline Sources/AfluentTesting/WaitUntilScheduled.swift:12]: Let's consider naming. This doesn't actually wait until a task is scheduled, it waits until it starts executing

**Review by Tyler-Keith-Thompson — APPROVED** [review]

File: Sources/AfluentTesting/WaitUntilScheduled.swift (hunk 2 above)
Comment (aim2120) [reply Sources/AfluentTesting/WaitUntilScheduled.swift:12]: Sure thing, will address this later today.

Comment (aim2120) [reply Sources/AfluentTesting/WaitUntilScheduled.swift]: Renamed to `waitUntilExecutionStarted`, how does that sound?

Comment (Tyler-Keith-Thompson) [reply Sources/AfluentTesting/WaitUntilScheduled.swift]: Perfect
